libs/labelbox/src/labelbox/schema/disconnected_task.py

```diff
@@ -0,0 +1,93 @@
+from typing import Any, Dict, List, Optional, Protocol, Union
+
+from labelbox.exceptions import OperationNotSupportedException, ResourceNotFoundError
+from labelbox.orm.db_object import experimental
+from labelbox.orm.model import Field
+from labelbox.schema.task import Task, DataUpsertTask
+
+TaskDataType = Optional[Union[Dict[str, Any], List[Dict[str, Any]]]]
+
+
+class SupportsTaskQueries(Protocol):
+
+    @property
+    def status(self) -> Field:
+        ...
+
+    @property
+    def errors(self) -> TaskDataType:
+        ...
+
+    @property
+    def result(self) -> TaskDataType:
+        ...
+
+
+class TaskFactory:
+
+    @classmethod
+    def get(cls, client, task_id) -> SupportsTaskQueries:
+        return cls.get_task_by_id(client, task_id)
+
+    @classmethod
+    def get_task_by_id(cls, client, task_id) -> SupportsTaskQueries:
+        user = client.get_user()
+        query = """
+            query GetUserCreatedTasksPyApi($userId: ID!, $taskId: ID!) {
+            user(where: {id: $userId}) {
+                createdTasks(where: {id: $taskId} skip: 0 first: 1) {
+                    completionPercentage
+                    createdAt
+                    errors
+                    metadata
+                    name
+                    result
+                    status
+                    type
+                    id
+                    updatedAt
+                }
+            }
+        }
+        """
+        result = client.execute(query, {"userId": user.uid, "taskId": task_id})
+        data = result.get("user", {}).get("createdTasks", [])
+        if not data:
+            raise ResourceNotFoundError(
+                message=f"The task {task_id} does not exist.")
+        task_data = data[0]
+        if task_data["type"].lower() == 'adv-upsert-data-rows':
+            task = DataUpsertTask(client, task_data)
+        else:
+            task = Task(client, task_data)
+
+        task._user = user
+        return task
+
+
+class DisconnectedTask:
```

Comment:
What's the value add of `DisconnectedTask`?

The value add is being able to surface a `Task` by `task_id` which is provided by `TaskFactory`.

Personally, I would avoid adding an additional type if it's only purpose is delegation.